### DISCUSSION TITLE
Something happened to cfl_box.cpp.o since 1.0.18?

### DISCUSSION THREAD

**Discussion by eadf** [fltk-rs/fltk-rs#772] [General]
I'm intermittently getting build error like this when using anything above 1.0.18.
Usually a `cargo clean` fixes it, but not always. `=1.0.18` (and earlier) is rock solid.   
```
  = note: ld: warning: directory not found for option '-L/Users/eadf/Develop/rust-workspace/boostvoronoi.rs/target/debug/build/fltk-sys-9fde8d298922d283/out/build'
          ld: warning: directory not found for option '-L/Users/eadf/Develop/rust-workspace/boostvoronoi.rs/target/debug/build/fltk-sys-9fde8d298922d283/out/build/Release'
          ld: warning: directory not found for option '-L/Users/eadf/Develop/rust-workspace/boostvoronoi.rs/target/debug/build/fltk-sys-9fde8d298922d283/out/lib64'
          ld: warning: directory not found for option '-L/Users/eadf/Develop/rust-workspace/boostvoronoi.rs/target/debug/build/fltk-sys-9fde8d298922d283/out/lib/Release'
          ld: warning: directory not found for option '-L/Users/eadf/Develop/rust-workspace/boostvoronoi.rs/target/debug/build/fltk-sys-9fde8d298922d283/out/lib64/Release'
          ld: in /Users/eadf/Develop/rust-workspace/boostvoronoi.rs/target/debug/deps/libfltk_sys-d4bbcac9ac633174.rlib(cfl_box.cpp.o), archive member 'cfl_box.cpp.o' with length 18784 is not mach-o or llvm bitcode for architecture arm64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
When it fails it always fails at cfl_box.cpp.o.

I'm switching between +stable and +nightly a lot for tests, could that be an issue?

Issue appears on M1 Mac.

Any ideas what could be the issue?

**Comment by MoAlyousef** [comment]
Since 1.0.18, libc was updated to 0.2.97, some locking code was refactored into its own header cfl_lock.hpp. The header is included in every source file. I think cfl_box.cpp.o appears always is just because of the alphabetical order of the object files, if there were a cfl_awidget.cpp the error message would be for that file. The 'directory not found` warning is harmless. Searching for similar messages, I found this issue:
https://github.com/alexcrichton/cc-rs/issues/542
The linked issues in it all seem to have similar error messages. I think shifting toolchains passes certain flags to cmake (the crate) which passes them to cc. 
Another similar message:
https://github.com/tauri-apps/tauri/issues/1167

Can you post the outputs of `cargo build -vv` using stable and nightly?

**Comment by eadf** [comment]
I ran the suggested script:

```
rm -rf /Library/Developer/CommandLineTools 
xcode-select --install
```
And since then I have not been able to reproduce the issue. 
I'll post build logs if the issue appears again.
Thanks for the links!
